docs/examples/couchbase.md

```diff
@@ -0,0 +1,236 @@
+
+### Setup credentials for the Couchbase server
+
+Sets the username and password credentials that will subsequently be used for managing the Couchbase server.
+
+```
+curl -v -X POST http://192.168.99.100:8091/settings/web -d port=8091 -d username=Administrator -d password=password
+* Hostname was NOT found in DNS cache
+*   Trying 192.168.99.100...
+* Connected to 192.168.99.100 (192.168.99.100) port 8091 (#0)
+> POST /settings/web HTTP/1.1
+> User-Agent: curl/7.37.1
+> Host: 192.168.99.100:8091
+> Accept: */*
+> Content-Length: 50
+> Content-Type: application/x-www-form-urlencoded
+> 
+* upload completely sent off: 50 out of 50 bytes
+< HTTP/1.1 200 OK
+* Server Couchbase Server is not blacklisted
+< Server: Couchbase Server
+< Pragma: no-cache
+< Date: Wed, 25 Nov 2015 22:50:43 GMT
+< Content-Type: application/json
+< Content-Length: 44
+< Cache-Control: no-cache
+< 
+* Connection #0 to host 192.168.99.100 left intact
+{"newBaseUri":"http://192.168.99.100:8091/"}
+```
+
+The command shows an HTTP POST request to the REST endpoint `/settings/web`. The user name and password credentials are passed in the request.
+
+### Install sample data
+
+The Couchbase server can be easily load some sample data in the Couchbase instance.
+
+```
+curl -v -u Administrator:password -X POST http://192.168.99.100:8091/sampleBuckets/install -d '["travel-sample"]'
+* Hostname was NOT found in DNS cache
+*   Trying 192.168.99.100...
+* Connected to 192.168.99.100 (192.168.99.100) port 8091 (#0)
+* Server auth using Basic with user 'Administrator'
+> POST /sampleBuckets/install HTTP/1.1
+> Authorization: Basic QWRtaW5pc3RyYXRvcjpwYXNzd29yZA==
+> User-Agent: curl/7.37.1
+> Host: 192.168.99.100:8091
+> Accept: */*
+> Content-Length: 17
+> Content-Type: application/x-www-form-urlencoded
+> 
+* upload completely sent off: 17 out of 17 bytes
+< HTTP/1.1 202 Accepted
+* Server Couchbase Server is not blacklisted
+< Server: Couchbase Server
+< Pragma: no-cache
+< Date: Wed, 25 Nov 2015 22:51:51 GMT
+< Content-Type: application/json
+< Content-Length: 2
+< Cache-Control: no-cache
+< 
+* Connection #0 to host 192.168.99.100 left intact
+[]
+```
+
+The command shows an HTTP POST request to the REST endpoint `/sampleBuckets/install`. The name of the sample bucket is passed in the request.
+
+Congratulations, you are now running a Couchbase container, fully configured using the REST API.
+
+## Query Couchbase using CBQ
+
+[CBQ](http://developer.couchbase.com/documentation/server/4.1/cli/cbq-tool.html), short for Couchbase Query, is a CLI tool that allows to create, read, update, and delete JSON documents on a Couchbase server. This tool is installed as part of the Couchbase Docker image.
+
+Run CBQ tool:
+
+```
+docker run -it --link db:db couchbase cbq --engine http://192.168.99.100:8093
```

Comment:
you can use `http://db:8093` here; containers can communicate with each other through the internal container-container network